airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ReplicationView.tsx

```diff
@@ -114,59 +99,62 @@ export const ReplicationView: React.FC<ReplicationViewProps> = ({ onAfterSaveSch
     if (activeUpdatingSchemaMode) {
       setActiveUpdatingSchemaMode(false);
     }
-
-    formikHelpers?.resetForm({ values });
-  };
-
-  const openResetDataModal = (values: ValuesProps) => {
-    openConfirmationModal({
-      title: "connection.updateSchema",
-      text: "connection.updateSchemaText",
-      submitButtonText: "connection.updateSchema",
-      submitButtonDataId: "refresh",
-      onSubmit: async () => {
-        await onSubmit(values);
-        closeConfirmationModal();
-      },
-    });
   };
 
   const onSubmitForm = async (values: ValuesProps) => {
-    if (activeUpdatingSchemaMode) {
-      openResetDataModal(values);
+    console.log("values.syncCatalog", values.syncCatalog);
+    console.log("initialConnection", initialConnection.syncCatalog);
+    const hasCatalogChanged = !equal(values.syncCatalog, initialConnection.syncCatalog);
+
+    console.log("hasCatalogChanged?", hasCatalogChanged);
+    if (hasCatalogChanged) {
+      const stateType = await connectionService.getStateType(connectionId);
+      console.log("ConnectionStateType", stateType);
+      if (stateType === ConnectionStateType.legacy) {
+        // The state type is legacy so the server will do a full reset after saving
+        // TODO: Show confirm dialog with full reset option
+        // TODO: This should render a nicer modal with a checkbox inside, once we have a proper modal service
+        const result = await openModal({ title: "test", content: () => <h2>props</h2> });
+        console.log("result", result);
+        if (result.type !== "canceled") {
+          // TODO: right skipRefresh
+          await saveConnection(values, false);
+        }
+        // openConfirmationModal({
+        //   title: "connection.updateSchema",
+        //   text: "connection.updateSchemaText",
+        //   submitButtonText: "connection.updateSchema",
+        //   submitButtonDataId: "refresh",
+        //   secondaryButtonText: "connection.updateSchemaWithoutReset",
+        //   onSubmit: async (type) => {
+        //     await saveConnection(values, type === "secondary");
+        //     closeConfirmationModal();
+        //   },
+        // });
+      } else {
+        // TODO: Show confirm dialog with partial reset option
+        // TODO: This should render a nicer modal with a checkbox inside, once we have a proper modal service
+        const result = await openModal({ title: "test", content: () => <h2>props</h2> });
+        console.log("result", result);
+      }
     } else {
-      await onSubmit(values);
+      // The catalog hasn't changed. We don't need to ask for any confirmation and can simply save
+      // TODO: Clarify if we want to have `skipRefresh` true or false in this case with BE.
```

Comment:
I think we probably want this to set `skipReset` to `true` in this case. The reason being that we won't ask for any confirmation, so we don't want to accidentally cause a reset to happen if, for example, another user changed the configuration as this request is being sent to the backend. If `skipReset` was false, that scenario would result in use calculating the diff and finding a change, and submitting a reset, even though we never prompted the user for confirmation of this.

Setting `skipReset` to `true` here guarantees that when we don't ask the user for confirmation, then that save will not cause a reset to be submitted

Reply:
Thanks. This would also be my preferred way, to never accidentally reset data.